core/validation/initial_check.py

```diff
@@ -0,0 +1,94 @@
+that the input for these is valid for the specific round of data we are ingesting.
+"""
+import numpy as np
+import pandas as pd
+
+import core.validation.failures as vf
+from core.const import EXPECTED_ROUND_THREE_SHEETS
+
+
+def extract_round_three_submission_details(workbook: dict[str, pd.DataFrame]) -> dict[str, list[str]]:
+    """
+    Extract submission details for round three from the given workbook.
+
+    :param workbook: A dictionary where keys are sheet names and values are pandas DataFrames.
+    :return: A dictionary containing the extracted submission details.
+    """
+    details_dict = {}
+
+    missing_sheets = check_missing_sheets(EXPECTED_ROUND_THREE_SHEETS, workbook)
+
+    if len(missing_sheets) > 0:
+        return {"Missing Sheets": missing_sheets}
+
+    sheet_a1 = workbook.get("1 - Start Here")
+    if sheet_a1 is not None:
+        details_dict["Form Version"] = [
+            sheet_a1.iloc[6][1],
+            "Town Deals and Future High Streets Fund Reporting Template (v3.0)",
+        ]
+        details_dict["Reporting Period"] = [sheet_a1.iloc[4][1], "1 October 2022 to 31 March 2023"]
+
+    sheet_a2 = workbook.get("2 - Project Admin")
+    if sheet_a2 is not None:
+        details_dict["Fund Type"] = [sheet_a2.iloc[5][4], "Town_Deal", "Future_High_Street_Fund"]
+        details_dict["Place Name"] = [sheet_a2.iloc[6][4]]
+
+    return details_dict
+
+
+def pre_transformation_check(submission_details: dict[str, list[str]]) -> list[vf.ValidationFailure]:
+    """
+    Perform pre-transformation checks on the given workbook.
+
+    :param submission_details: A dictionary containing submission details.
+    :return: A list of validation failures encountered during pre-transformation checks.
+    """
+
+    if "Missing Sheets" in submission_details:
+        return [vf.EmptySheetFailure(empty_sheet=empty_sheet) for empty_sheet in submission_details["Missing Sheets"]]
+
+    pre_transformation_failures = [
+        check_form_input(key, value)
+        for key, value in submission_details.items()
+        if check_form_input(key, value) is not None
+    ]
+
+    return pre_transformation_failures
+
+
+def check_form_input(
+    value_descriptor: str, values: list[str]
+) -> vf.NoInputFailure | vf.PreTransformationFailure | None:
+    """
+    Check the form input for pre-transformation failures.
+
+    :param value_descriptor: A string describing the form input value.
+    :param values: A string containing the form input values.
+    :return: A ValidationFailure object representing the failure, if any.
+    """
+    if values[0] in ["", np.nan]:
+        return vf.NoInputFailure(value_descriptor=value_descriptor)
+
+    if value_descriptor != "Place Name" and values[0] not in values[1:]:
+        return vf.PreTransformationFailure(
+            value_descriptor=value_descriptor, entered_value=values[0], expected_values=values[1:]
+        )
```

Comment:
Couple of points:

- Shouldn't combine the entered value and expected into a single list of strings like this. 
- The null check should be a separate function that is only used on a separate list of values because otherwise we have to make hard coded checks like the one for "Place Name".

I'd consider changing the submission_details to this structure:
```
submission_details = {
    "Missing Sheets": [],
    "NullChecks": {
        "Place Name": "Some Place not an Empty String",  #  this would pass the check
        "Something else": ""  # this would fail
    },
    "ValueChecks": {
        "Fund Type": ("TD", {"TD", "HS"}),  # a tuple of two values, the entered, and the set of expected
        "Something else": ("Entered", {"Expected1", "Expected2"})
    }
}
```
This means we can iterate over the null checks and call the null check function for those, and for the value checks we can call this function.

A dictionary will do for now but we'll probably want to make this an object to better define and enforce the structure at some point.

Reply:
I've changed the structure to look like this.

Was thinking we could actually make the extraction spreadsheet agnostic if the expected_values for each round were tied to an object in const, and then we could just add source_type as a param and wouldn't need to add any code for subsequent rounds, but Round 1 and Round 2 won't have any pre-transformation validations and we can't be sure about the specific cell location in subsequent rounds so I've stuck with that you've suggested.